x-pack/legacy/plugins/apm/public/components/shared/TransactionActionMenu/TransactionActionMenu.tsx

```diff
@@ -4,240 +4,83 @@
-        { defaultMessage: 'Show host metrics' }
-      ),
-      condition: !!hostName,
-      path: `/link-to/host-detail/${hostName}`,
-      query: infraMetricsQuery
-    }
-  ];
-
-  const infraItems = infraConfigItems.map(
-    ({ icon, label, condition, path, query }, index) => ({
-      icon,
-      key: `infra-link-${index}`,
-      child: (
-        <InfraLink path={path} query={query}>
-          {label}
-        </InfraLink>
-      ),
-      condition
-    })
-  );
+  const [isOpen, setIsOpen] = useState(false);
 
-  const uptimeLink = url.format({
-    pathname: core.http.basePath.prepend('/app/uptime'),
-    hash: `/?${fromQuery(
-      pick(
-        {
-          dateRangeStart: urlParams.rangeFrom,
-          dateRangeEnd: urlParams.rangeTo,
-          search: `url.domain:"${transaction.url?.domain}"`
-        },
-        (val: string) => !!val
-      )
-    )}`
+  const sections = getSections({
+    transaction,
+    basePath: core.http.basePath,
+    location,
+    urlParams
   });
 
-  const menuItems = [
-    ...infraItems,
-    {
-      icon: 'discoverApp',
-      key: 'discover-transaction',
-      condition: true,
-      child: (
-        <DiscoverTransactionLink transaction={transaction}>
-          {i18n.translate(
-            'xpack.apm.transactionActionMenu.viewSampleDocumentLinkLabel',
-            {
-              defaultMessage: 'View sample document'
-            }
-          )}
-        </DiscoverTransactionLink>
-      )
-    },
-    {
-      icon: 'uptimeApp',
-      key: 'uptime',
-      child: (
-        <EuiLink href={uptimeLink}>
-          {i18n.translate('xpack.apm.transactionActionMenu.viewInUptime', {
-            defaultMessage: 'View monitor status'
-          })}
-        </EuiLink>
-      ),
-      condition: transaction.url?.domain
-    }
-  ]
-    .filter(({ condition }) => condition)
-    .map(({ icon, key, child }) => (
-      <EuiContextMenuItem icon={icon} key={key}>
-        <EuiFlexGroup gutterSize="s">
-          <EuiFlexItem>{child}</EuiFlexItem>
-          <EuiFlexItem grow={false}>
-            <EuiIcon type="popout" />
-          </EuiFlexItem>
-        </EuiFlexGroup>
-      </EuiContextMenuItem>
-    ));
-
   return (
-    <EuiPopover
+    <ActionMenu
       id="transactionActionMenu"
-      button={<ActionMenuButton onClick={() => setIsOpen(!isOpen)} />}
-      isOpen={isOpen}
       closePopover={() => setIsOpen(false)}
+      isOpen={isOpen}
       anchorPosition="downRight"
-      panelPaddingSize="none"
+      button={<ActionMenuButton onClick={() => setIsOpen(!isOpen)} />}
     >
-      <EuiContextMenuPanel
-        items={menuItems}
-        title={i18n.translate('xpack.apm.transactionActionMenu.actionsLabel', {
-          defaultMessage: 'Actions'
-        })}
-      />
-    </EuiPopover>
+      {sections.map((section, idx) => (
+        <div key={idx}>
+          {section.map(item => (
+            <Section key={item.key} marginBottom={px(units.plus)}>
```

Comment:
No need to add a marginBottom here, should just use the `<SectionSpacer />` component https://github.com/elastic/kibana/blob/master/x-pack/plugins/observability/public/components/action_menu.tsx#L46

Reply:
Thanks for your feedback @phillipb, I added the `marginBottom` property because I thought the Section component should automatically handle the space between sections for me. Then with CSS I remove the space from the last item.

https://github.com/cauemarcondes/kibana/blob/actions-menu/x-pack/plugins/observability/public/components/action_menu.tsx#L49-L56

Reply:
ah, I see. That makes sense. What do you think about defaulting the `marginBottom` in the Section component, so others don't need to think about it?

Reply:
They don't have to worry about it, the property is optional if it is missing it will use 0.

Reply:
I just meant, so that they get this behavior by default. This feels like the desired behavior to me, or is this APM specific?

Reply:
I think the following is more readable:

```ts
<Section key={item.key}>
!isLast &&  <SectionSpacer />
```

Although using css is more performant... 🤔

Reply:
In the javascript approach it's clear that we add a space after every section except the last one. That's not possible to see from:
```
<Section key={item.key} marginBottom={px(units.plus)}>
```

Reply:
I don't have strong feelings about one approach over the other. I just think if we go with the css approach it should all be handled internally in the Section component. A `marginBottom` prop feels like an afterthought. Some future person will wonder why it's there and what they should set it to. I'd rather keep the api clean.

Reply:
IMO, I expect the Service Component to handle that. Otherwise, what's the benefit that I have by using it? In the end, it just shows the `children`. Now if I start using it in other parts of the UI, every time I'd have to check if I'm in the last item of the list to not show the space. I agreed with @phillipb that we could add the margin-bottom by default in order to have a clear API. WDYT?

Reply:
> we could add the margin-bottom by default in order to have a clear API
👍